includes/functions.php

```diff
@@ -2286,22 +2296,15 @@ function tsml_scan_data_source($data_source_url)
                 list($import_meetings, $delete_meeting_ids,  $change_log) = tsml_get_changed_import_meetings($meetings, $data_source_url, $data_source_parent_region_id);
                 if (is_array($change_log) && count($change_log)) {
                     // Send Email notifying Admins that this Data Source needs updating
-                    $message = "Data Source changes were detected during a scheduled sychronization check with this feed: $data_source_url. Your website meeting list details based on the $data_source_name feed are no longer in sync. <br><br>Please sign-in to your website and refresh the $data_source_name Data Source feed found on the Meetings Import & Export page.<br><br>";
-                    $message .= "data_source_name: $data_source_name <br>";
-                    $term = get_term_by('term_id', $data_source_parent_region_id, 'tsml_region');
-                    $parent_region = $term->name;
-                    $message .= "parent_region: $parent_region <br>";
-                    $message .= "database record count: $data_source_count_meetings <br>";
-                    $feedCount = count($meetings);
-                    $message .= "data source feed count: $feedCount<br>";
-                    $message .= "Last Refresh: <span style='color:red;'>*</span>" . Date("l F j, Y  h:i a", $data_source_last_import) . '<br>';
-                    $message .= "<br><b><u>Detected Difference</b></u><br>";
-                    $message .= tsml_build_import_change_report($data_source_name, $change_log);
-                    $import_page_url = admin_url('edit.php?post_type=tsml_meeting&page=import');
-                    $message .= "<a href='" . $import_page_url . "' style=' margin: 0 auto;background-color: #4CAF50;border: none;color: white;padding: 25px 32px;text-align: center;text-decoration: none;display: block;font-size: 18px;'>Go to Import & Settings page</a>";
+                    $message = "<br>Please sign in to your website and refresh the <b>$data_source_name</b> feed on the Import & Export page.<br><br>";
+                    $message .= tsml_build_import_change_report($data_source_name, $change_log, true);
+                    $refresh_form_name = tsml_make_refresh_attr_name($data_source_name, 'frm');
+                    $import_page_url = admin_url('/edit.php?post_type=tsml_meeting&page=import');
+		    $button_text = __("Refresh $data_source_name Data Source", '12-step-meeting-list');
```

Comment:
can't have variables inside localizable strings - might be easier to take out the variable here? otherwise you want to use `sprintf` eg

```
sprintf(__('Please upload a csv file. Your file ended in .%s.', '12-step-meeting-list'), $extension);
```

Reply:
Why not?

Reply:
Ok

Reply:
Thanks. I've added that to my toolkit.